contents/javascript/staticTypingWithTypescript.md

```diff
@@ -0,0 +1,54 @@
+<frontmatter>
+  title: Static Typing with TypeScript
+  header: pagetop.md
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+<div class="website-content">
+
+{{ booktitle | safe }}
+
+# Static Typing with TypeScript
+
+**Author(s): [Lu Yang Kenneth](https://github.com/luyangkenneth)**
+
+Reviewer(s): WIP FILL THIS OUT LATER
+
+
+## What is Static Typing?
+
+**Statically typed languages** like Java, Go, and C++ are able to catch type-related errors at compile time, while in **dynamically typed languages** like JavaScript, such errors are not as easily discoverable because the types of variables are only known at runtime.
+
+To help identify these problems without having to run any code, many developers have adopted technologies like Flow and TypeScript, which enable static type checking for codebases written in JavaScript.
+
+Equivalent type checking tools exist for other dynamically typed languages like Python and Ruby as well, which suggests that this issue is not specific to the JavaScript community. The decision to move towards a statically typed codebase is usually motivated by the following benefits:
+
+- Static type checking **reduces type-related errors during the development process**. These errors can be expensive to fix if they leak into production and cause downtime in your application, especially for teams that value stability over short-term productivity. A type checker provides an automatic way to verify the type safety and correctness of your application _before_ it goes live.
```

Comment:
I think there is a gap between writing code (where TS catches errors) and going live, so perhaps you might be able to be more specific when discussing when exactly type errors can be eliminated.

Also, correctness can also be determined via tests and manual testing during the development stage as well, but those are much costlier and less reliable than static tools. Perhaps that is something you might want to discuss here as well! 🙂

Reply:
Oooh yeah good to make those other related things explicit